File: smashpen/src/main/java/com/smashpen/smashpen/domain/SmashCharacter.java
package com.smashpen.smashpen.domain;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class SmashCharacter {

	@Id @GeneratedValue private Long id;

	@Column(unique=true) private String name;

	SmashCharacter() {}

	public SmashCharacter(String name) {
		this.name = name.toLowerCase();
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
}

File: smashpen/src/main/java/com/smashpen/smashpen/domain/SmashCharacter.kt
package com.smashpen.smashpen.domain

import javax.persistence.*

@Entity
class SmashCharacter {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = -1

    var name: String? = null

    constructor(name: String) {
        this.name = name.toLowerCase()
    }
}
